setup_build.py

```diff
@@ -117,6 +120,20 @@ def _make_extensions(cls, config):
         if os.name != 'nt':
             settings['runtime_library_dirs'] = settings['library_dirs']
 
+        for module in ALL_MODULES:
+            raw_path = Path(localpath("h5py")).joinpath(module).resolve()
+            for ext in ['.pyx', '.pxd', '.pxi']:
+                if not (templ := raw_path.with_suffix(f'.templ{ext}')).exists():
+                    continue
+
+                if (target := raw_path.with_suffix(ext)).exists():
+                    current_text = target.read_text()
+                else:
+                    current_text = ""
+                new_text = tempita.sub(templ.read_text(), **templ_config)
+                if new_text != current_text:
+                    target.write_text(new_text)
```

Comment:
```suggestion
                    current_text = target.read_text('utf-8')
                else:
                    current_text = ""
                new_text = tempita.sub(templ.read_text(), **templ_config)
                if new_text != current_text:
                    target.write_text(new_text, 'utf-8')
```

The default encoding is unfortunately platform dependent - IMO a missed opportunity when Python 3 came out.

Maybe everything in our Cython files is ASCII anyway, in which case the default will work, but it's easier to do it right now than figure out a build problem on Windows later.

Reply:
Good point. Thank you !